File: breakPalindrome.py
#Approach 1- exactly like .toCharArray()
def breakPalindrome(palindrome):
	s=list(palindrome)
	if len(palindrome)==1: return ""
	for i in range(len(palindrome)//2):
	    if s[i]!='a':
	        s[i]='a'
	        return "".join(s)

	s[len(palindrome)-1]='b'
	return "".join(s)

#Approach 2- slicing 
def breakPalindrome1(palindrome):
	if len(palindrome) == 1:
            return ""
	for i in range(len(palindrome) // 2):
	    if palindrome[i] != 'a':
	        return palindrome[:i] + 'a' + palindrome[i+1:]

	return palindrome[:-1] + 'b'

print(breakPalindrome("abba"))
print(breakPalindrome1("aaaa"))

File: breakPalindrome.java
class Solution{
	public String breakPalindrome(String palindrome){
		
		// Strings are immutable in Java, convert it into a char array
		char[] palindromeArray= palindrome.toCharArray();
		for (int i=0; i<palindrome.length()/2; i++){
			if(palindromeArray[i]!='a'){
				palindromeArray[i]='a';
				return String.valueOf(palindromeArray);
			}
		}
		palindromeArray[palindrome.length()-1]='b';
		return String.valueOf(palindromeArray);
	}
}

class breakPalindrome{
	public static void main(String[] args) {
		String palindrome="abba";
		Solution s= new Solution();
		System.out.println(s.breakPalindrome(palindrome));

	}
}